framework/src/utils/MooseUtils.C

```diff
@@ -54,10 +54,13 @@ escape(std::string & str)
 }
 
 std::string
-trim(std::string str, const std::string & white_space)
+trim(const std::string & str, const std::string & white_space)
```

Comment:
I guess you didn't like it modifying the string in place. I suppose this is fine.

Reply:
The old version wasn't working quite right and there was no reason to copy.